File: dotnet/sdk/Files.com/Models/Preview.cs

using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text.Json;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Files.com.Models
{
    public class Preview
    {
        private Dictionary<string, object> attributes;
        private Dictionary<string, object> options;

        public Preview()
        {
            this.attributes = new Dictionary<string, object>();
            this.options = new Dictionary<string, object>();

            this.attributes.Add("id", null);
            this.attributes.Add("status", null);
            this.attributes.Add("download_uri", null);
            this.attributes.Add("type", null);
            this.attributes.Add("size", null);
        }

        public Preview(Dictionary<string, object> attributes, Dictionary<string, object> options)
        {
            this.attributes = attributes;
            this.options = options;
        }

        /// <summary>
        /// Preview ID
        /// </summary>
        [JsonPropertyName("id")]
        public Nullable<Int64> Id
        {
            get { return (Nullable<Int64>) attributes["id"]; }
        }

        /// <summary>
        /// Preview status.  Can be invalid, not_generated, generating, complete, or file_too_large
        /// </summary>
        [JsonPropertyName("status")]
        public string Status
        {
            get { return (string) attributes["status"]; }
        }

        /// <summary>
        /// Link to download preview
        /// </summary>
        [JsonPropertyName("download_uri")]
        public string DownloadUri
        {
            get { return (string) attributes["download_uri"]; }
        }

        /// <summary>
        /// Preview status.  Can be invalid, not_generated, generating, complete, or file_too_large
        /// </summary>
        [JsonPropertyName("type")]
        public string Type
        {
            get { return (string) attributes["type"]; }
        }

        /// <summary>
        /// Preview size
        /// </summary>
        [JsonPropertyName("size")]
        public Nullable<Int64> Size
        {
            get { return (Nullable<Int64>) attributes["size"]; }
        }



    }
}

File: java/src/main/java/com/files/models/Preview.java

package com.files.models;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.HashMap;

public class Preview
{
    private HashMap<String, Object> attributes;
    private HashMap<String, Object> options;

    public Preview(HashMap<String, Object> attributes, HashMap<String, Object> options)
    {
        this.attributes = attributes;
        this.options = options;
    }

    /**
     * Preview ID
     */
    @Getter
    public Long id;

    /**
     * Preview status.  Can be invalid, not_generated, generating, complete, or file_too_large
     */
    @Getter
    public String status;

    /**
     * Link to download preview
     */
    @Getter
    public String downloadUri;

    /**
     * Preview status.  Can be invalid, not_generated, generating, complete, or file_too_large
     */
    @Getter
    public String type;

    /**
     * Preview size
     */
    @Getter
    public Long size;



}


